src/components/vm/disks/diskAdd.tsx

```diff
@@ -808,7 +960,7 @@ const AddDiskModalFooter = ({
                     variant='primary'
                     isLoading={addDiskInProgress || verificationInProgress}
                     isDisabled={addDiskInProgress || verificationInProgress || dialogLoading ||
-                                (storagePools.length == 0 && mode != CUSTOM_PATH) ||
+                                ((storagePools || []).length == 0 && mode != CUSTOM_PATH) ||
```

Comment:
Same question as above.

Reply:
This is about the local `storagePools` variable, which is undefined until the list of pools has been retrieved.  In fact, when `dialogLoading` is false, storagePools is guaranteed to be defined...

So, hmm, I would say this is fine, if a bit obscure.